avoinspector/src/development/java/app/avo/inspector/VisualInspector.java

```diff
@@ -0,0 +1,112 @@
+package app.avo.inspector;
+
+import android.app.Activity;
+import android.app.Application;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import app.avo.androidanalyticsdebugger.DebuggerManager;
+import app.avo.androidanalyticsdebugger.DebuggerMode;
+import app.avo.androidanalyticsdebugger.EventProperty;
+import app.avo.androidanalyticsdebugger.PropertyError;
+
+public class VisualInspector {
+    @Nullable
+    DebuggerManager debugger;
+
+    public VisualInspector(@NonNull AvoInspectorEnv env, @NonNull Application application, @Nullable Activity rootActivityForVisualInspector) {
+        if (env != AvoInspectorEnv.Prod) {
+            debugger = new DebuggerManager(application);
+            if (rootActivityForVisualInspector != null) {
+                show(rootActivityForVisualInspector, VisualInspectorMode.BUBBLE);
+            }
+        }
+    }
+
+    public @Nullable
+    Object getDebuggerManager() {
+        return debugger;
+    }
```

Comment:
I've never seen the annotation being written like this. This is how it's used usually:
```suggestion
    @Nullable
    public Object getDebuggerManager() {
        return debugger;
    }
```